source/adapters/hip/command_buffer.cpp

```diff
@@ -122,6 +122,25 @@ ur_result_t UR_APICALL urCommandBufferAppendMemBufferReadRectExp(
   return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
 }
 
+UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendUSMPrefetchExp(
+    ur_exp_command_buffer_handle_t, const void *, size_t,
+    ur_usm_migration_flags_t, uint32_t,
+    const ur_exp_command_buffer_sync_point_t *,
+    ur_exp_command_buffer_sync_point_t *) {
+  detail::ur::die("Experimental Command-buffer feature is not "
+                  "implemented for HIP adapter.");
+  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendUSMAdviseExp(
+    ur_exp_command_buffer_handle_t, const void *, size_t, ur_usm_advice_flags_t,
+    uint32_t, const ur_exp_command_buffer_sync_point_t *,
+    ur_exp_command_buffer_sync_point_t *) {
+  detail::ur::die("Experimental Command-buffer feature is not "
+                  "implemented for HIP adapter.");
+  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
```

Comment:
Ideally we would like to move away from the use of `die` `assert` `terminate`.
Also this is for processing/appending a driver hint command which being just a hint shouldn't terminate/crash the end application, hence resulting in a success and ignoring may be acceptable here instead.
Another approach if you would want to warn the end user about what is happening is to trigger handling this as a warning and print to stderr/out which is currently done through the not so handy mechanism using `setErrorMessage` with `UR_RESULT_SUCCESS` error code and then return `UR_RESULT_ADAPTER_SPECIFIC_ERROR` which the runtime can handle as either an error or a warning if `setErrorMessage` was set with success.

Reply:
Thanks for this detailed feedback. We added a warning in CUDA support for prefetch and mem-advice hints based on your advice. 
However, there is currently no support for CommandBuffer in the Hip backend and even though these operations are just hints, we still need to handle the associated dependencies for correct graph execution. So we prefer that these functions  return `UNSUPPORTED_FEATURE` and terminate with an error until Hip support for CommandBuffer is implemented.

Reply:
Yeah I understand! Thanks for your explanation, obviously I overlooked the whole CommandBuffer implementation picture. This looks fine to me then.